src/argilla/client/feedback/integrations/huggingface/dataset.py

```diff
@@ -160,10 +162,10 @@ def _huggingface_format(dataset: Union["FeedbackDataset", "RemoteFeedbackDataset
             hf_dataset["external_id"].append(record.external_id or None)
 
             if vectors_settings:
+                vectors = {}
                 for vector_settings in vectors_settings:
-                    if vector_settings.name not in hf_dataset:
-                        hf_dataset[vector_settings.name] = []
-                    hf_dataset[vector_settings.name].append(record.vectors.get(vector_settings.name, None))
+                    vectors.update({vector_settings.name: record.vectors.get(vector_settings.name, None)})
```

Comment:
Do we need to provide the same dict structure for vectors in HF for ALL the records? I mean, having 2 vectors,  `v1` and `v2` and record[0] only with `v1`, and record[1] only with `v2`. Do we still need create a the whole dictionary even if there is partial info?

Reply:
No, we can skip it if `None`

Reply:
If you are skipping then later if they are `None` you can avoid them to be added on first instance too right?